homeassistant/components/powerwall/switch.py

```diff
@@ -0,0 +1,62 @@
+"""Support for Powerwall Switches (V2 API only)."""
+
+from typing import Any
+
+from tesla_powerwall import GridStatus, IslandMode
+
+from homeassistant.components.switch import SwitchDeviceClass, SwitchEntity
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity import EntityCategory
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from .const import DOMAIN
+from .entity import PowerWallEntity
+from .models import PowerwallRuntimeData
+
+ICON = "mdi:home-battery-outline"
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up Powerwall switch platform from Powerwall resources."""
+    powerwall_data: PowerwallRuntimeData = hass.data[DOMAIN][config_entry.entry_id]
+    async_add_entities([PowerwallOffGridEnabledEntity(powerwall_data)])
+
+
+class PowerwallOffGridEnabledEntity(PowerWallEntity, SwitchEntity):
+    """Representation of a Switch entity for Powerwall Off-grid operation."""
+
+    _attr_name = "Take Powerwall Off-Grid"
+    _attr_unique_id = "take_powerwall_off_grid"
```

Comment:
```suggestion
    _attr_name = "Powerwall Off-Grid"
    _attr_unique_id = "powerwall_off_grid"
```

Reply:
But the unique id won't work if they have multiple power wall systems since it needs to be set in `__init__` and include the base unique id

Reply:
I think `Off-Grid operation` might be a better name since its already bound to the powerwall device

Reply:
Pretty sure I've set this up as expected